eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1408 @@
+      g_autofree gchar *license_link = NULL;
+      g_autofree gchar *disclaimer = NULL;
+
+      if (original_link == NULL)
+        return NULL;
+
+      license_link = format_license_link (license, error);
+
+      if (license_link == NULL)
+        return NULL;
+
+      disclaimer = g_strdup_printf(_("This page conatins content from %s, available under a %s license."),
+                                   source, license_link);
+      return g_variant_new_string (disclaimer);
+    }
+  else if (g_strcmp0 (source, "wikihow") == 0)
+    {
+      g_autofree gchar *wikihow_article_link = format_a_href_link (original_uri,
+                                                                   title,
+                                                                   error);
+      g_autofree gchar *wikihow_link = NULL;
+      g_autofree gchar *disclaimer = NULL;
+
+      if (wikihow_article_link == NULL)
+        return NULL;
+
+      wikihow_link = format_a_href_link (_("http://wikihow.com"),
+                                         source_name,
+                                         error);
+
+      if (wikihow_link == NULL)
+        return NULL;
+
+      disclaimer = g_strdup_printf(_("See %s for more details, videos, pictures and attribution. Courtesy of %s, where anyone can easily learn how to do anything."),
+                                   wikihow_article_link,
+                                   wikihow_link);
+
+      return g_variant_new_string (disclaimer);
+    }
+
+  return g_variant_new_boolean (FALSE);
+}
+
+static GVariant *
+get_legacy_css_files (const gchar *source)
+{
+  const gchar * const empty_css_files[] = { NULL };
+
+  if (g_strcmp0 (source, "wikisource") == 0 ||
+      g_strcmp0 (source, "wikibooks") == 0 ||
+      g_strcmp0 (source, "wikipedia") == 0)
+    {
+      const gchar * const css_files[] = {
+        "wikimedia.css",
+        NULL
+      };
+      return g_variant_new_strv (css_files, -1);
+    }
+  else if (g_strcmp0 (source, "wikihow") == 0)
+    {
+      const gchar * const css_files[] = {
+        "wikihow.css",
+        NULL
+      };
+      return g_variant_new_strv (css_files, -1);
+    }
+
+  return g_variant_new_strv (empty_css_files, -1);
+}
+
+static GVariant *
+get_legacy_javascript_files (gboolean use_scroll_manager)
+{
+  GPtrArray *javascript_files = g_ptr_array_new_with_free_func (g_free);
+  g_auto(GStrv) strv = NULL;
+
+  g_ptr_array_add (javascript_files, g_strdup ("content-fixes.js"));
+  g_ptr_array_add (javascript_files, g_strdup ("hide-broken-images.js"));
+
+  if (use_scroll_manager)
+    g_ptr_array_add (javascript_files, g_strdup ("scroll-manager.js"));
+
+  /* NULL-terminate */
+  g_ptr_array_add (javascript_files, NULL);
+
+  strv = (GStrv) g_ptr_array_free (javascript_files, FALSE);
+  return g_variant_new_strv ((const gchar * const *) strv, -1);
+}
+
+static GVariant *
+get_legacy_should_include_mathjax (const gchar *source)
```

Comment:
I am a little torn as to whether these helper functions should be returning a native value or `GVariant *` directly. It is slightly more convenient if it is the latter, even if it couples it to the caller somewhat.